src/pypi_attestation_models/_cli.py

```diff
@@ -0,0 +1,302 @@
+    )
+
+    sign_command.add_argument(
+        "files",
+        metavar="FILE",
+        type=Path,
+        nargs="+",
+        help="The file to sign",
+    )
+
+    verify_command = subcommands.add_parser(
+        name="verify",
+        help="Verify one or more inputs",
+        parents=[parent_parser],
+    )
+
+    verify_command.add_argument(
+        "--identity",
+        type=str,
+        required=True,
+        help="Signer identity",
+    )
+
+    verify_command.add_argument(
+        "--staging",
+        action="store_true",
+        default=False,
+        help="Use the staging environment",
+    )
+
+    verify_command.add_argument(
+        "files",
+        metavar="FILE",
+        type=Path,
+        nargs="+",
+        help="The file to sign",
+    )
+
+    inspect_command = subcommands.add_parser(
+        name="inspect",
+        help="Inspect one or more inputs",
+        parents=[parent_parser],
+    )
+
+    inspect_command.add_argument(
+        "--dump-bytes",
+        action="store_true",
+        default=False,
+        help="Dump the bytes of the signature",
+    )
+
+    inspect_command.add_argument(
+        "files",
+        metavar="FILE",
+        type=Path,
+        nargs="+",
+        help="The file to sign",
+    )
+
+    return parser
+
+
+def _die(args: argparse.Namespace, message: str) -> NoReturn:
+    """Handle argument parsing errors and terminate the program.
+
+    Fix up the type hints on our use of `ArgumentParser.error`.
+    """
+    args._parser.error(message)  # noqa: SLF001.
+    raise ValueError("unreachable")
+
+
+def get_identity_token(args: argparse.Namespace) -> IdentityToken:
+    """Generate an Identity Token.
+
+    This method uses the following order of precedence:
+    - A token passed as an argument
+    - An ambient credential
+    - An OAuth-2 flow
+    """
+    # First, check if a token was supplied
+    if args.identity_token:
+        return IdentityToken(args.identity_token)
+
+    # Ambient credential detection
+    oidc_token = detect_credential()
+    if oidc_token is not None:
+        return IdentityToken(oidc_token)
+
+    # Finally, OAuth-2 Flow
+    if args.staging:
+        issuer = Issuer.staging()
+    else:
+        issuer = Issuer.production()
+
+    return issuer.identity_token()
+
+
+def _sign(args: argparse.Namespace) -> None:
+    """Sign the files passed as argument."""
+    try:
+        identity = get_identity_token(args)
+    except IdentityError as identity_error:
+        _die(args, f"Failed to detect identity: {identity_error}")
+
+    if args.staging:
+        signing_ctx = SigningContext.staging()
+    else:
+        signing_ctx = SigningContext.production()
+
+    with signing_ctx.signer(identity, cache=True) as signer:
+        for file_path in args.files:
+            _logger.debug(f"Signing {file_path}")
+
+            if not file_path.is_file():
+                _die(args, f"{file_path} is not a file.")
```

Comment:
We should check that the arguments are valid (in this case, that all files exist) near the top of the function, and fail early if any of the files do not exist. 

That way, if we get `[file1, file2, invalid_file3, file4]` as arguments then we fail without signing anything, instead of throwing on `invalid_file3` and leaving `file4` without signing, but `file1` and `file2` signed.

Reply:
Good catch, I refactored this part.